File: src/com/anathema_roguelike/entities/characters/foes/traits/Extraordinary.java
package com.anathema_roguelike.entities.characters.foes.traits;

import com.anathema_roguelike.main.utilities.Utils;
import com.anathema_roguelike.stats.characterstats.secondarystats.detection.senses.Sense;
import com.anathema_roguelike.stats.effects.AdditiveCalculation;

public class Extraordinary<T extends Sense> extends StatTrait<T> {

	public Extraordinary(Class<T> attribute) {
		super("Extraordinary: " + Utils.getName(attribute), attribute, AdditiveCalculation.fixed(10.0));
	}
}

File: src/com/anathema_roguelike/entities/characters/foes/traits/Extraordinary.scala
package com.anathema_roguelike
package entities.characters.foes.traits

import com.anathema_roguelike.main.utilities.Utils
import com.anathema_roguelike.stats.characterstats.secondarystats.SecondaryStat
import com.anathema_roguelike.stats.effects.AdditiveCalculation
import scala.reflect.runtime.universe._

class Extraordinary[T <: SecondaryStat : TypeTag]()
  extends StatTrait[T]("Extraordinary: " + Utils.getName(typeTagToClass[T]), AdditiveCalculation.fixed(10.0)) {
}